dataset/bucket.cpp

```diff
@@ -188,4 +190,37 @@ void append(const VariableView &var0, const VariableConstView &var1) {
     var0.replace_model(combine<Dataset>(var0, var1));
 }
 
+namespace histogram_detail {
+template <class Out, class Coord, class Weight, class Edge>
+using args = std::tuple<span<Out>, span<const Coord>, span<const Weight>,
+                        span<const Edge>>;
+}
+
+Variable histogram(const VariableConstView &data,
+                   const VariableConstView &binEdges) {
+  using namespace scipp::core;
+  using namespace histogram_detail;
+  auto hist_dim = binEdges.dims().inner();
+  const auto &[indices, dim, buffer] = data.constituents<bucket<DataArray>>();
+  if (!buffer.masks().empty())
+    throw std::runtime_error("Masked data cannot be histogrammed yet.");
```

Comment:
Can you clarify, for my understanding, which masks you are talking about here? Maybe with a concrete example? Would it be masks on detector pixels for instance? Is this just being delayed to another PR or is there some inherent difficulty causing you to just throw an error here? Or maybe it's a use case that would only happen rarely in practise?

Reply:
It would be an per-event masking. Masking detector pixels would be unrelated (and even preserved).